docs/design/chart_format.md

```diff
@@ -0,0 +1,664 @@
+* Distinguishing the development workflow from the release workflow at a location that is intuitive and common for developers
+* Providing a method for verifying immutability of a version by checksum
+* Removing vendor reliance
+* Separating the concept of code from the concept of release artifact
+
+## Chart References: Long form, short form, and local reference
+There are three reference forms for a chart. A fully qualified (long) form, a mnemonic (short) form, and a local path spec.
+### Long Form
+A long form chart reference is an exact URL to the chart. It should use HTTP or HTTPS protocols, and follow the typical rules of a complete URL: https://example.com/charts/nginx-1.2.3.tgz
+
+
+A long form reference must include a protocol spec. It may also contain any pattern allowed in URLs.
+
+
+The following URI schemes must be supported:
+* http:
+* https:
+
+
+#### Special Form Google Storage and S3
+The [Google Storage scheme](https://cloud.google.com/storage/docs/interoperability) (`gs://`)  and the S3 scheme (`s3://`) can also be support. These forms do not follow the URL standard (bucket name is placed in the host field), but their behavior is well-documented.
+
+
+Examples:
+```
+gs://charts/nginx-1.2.3.tgz
+s3://charts/nginx-1.2.3.tgz
+```
+### Short Form
+A short form package reference provides the minimum necessary information for accessing a chart. Short form URLs use a defined [URI scheme](https://tools.ietf.org/html/rfc3986#section-3.1) of `helm`:.
+
+
+A generic short form maps to an object storage reference (`DOMAIN/REPOSITORY/RELEASE`).
+
+```
+helm:example.com/charts/nginx-1.2.3.tgz
+```
+
+For the purpose of providing versioning ranges, and also for backward compatibility, version requirements may be treated as as suffix condition, instead of part of the path:
+
+```
+helm:example.com/charts/nginx#1.2.3   // Exact version
+helm:example.com/charts/nginx#~2.1    // Version range
+helm:example.com/charts/nginx         // Latest release
+```
+
+The first of the above two short names is equivalent to helm:example.com/charts/nginx-1.2.3.tgz. The second example uses a semantic version filter ~1.2, which means “>=1.2.0, <1.3.0”, or “any patch version of 1.2”. Other filters include the “^” operator (^1 is “any minor version in the 1.x line), and “>”, “>=”, “=”, “<”, and “<=” operators. These are standard SemVer filter operators.
+
+
+Any short form handler should be able to resolve the default short form as specified above.
+
+
+### Local References
+During chart development, and in other special circumstances, it may be desirable to work with an unversioned, unpackaged local copy of a chart. For the sake of consistency across products, an explicit naming formula is to be followed.
+
+
+In cases where a local path is used in lieu of a full or short name, the path string _must_ begin with either a dot (.) or a slash (/) or the file schema(`file`:).
```

Comment:
or the file **schema**(`file`:) -> or be a URL with the file **scheme** (`file:`).